app/services/invoices/update_service.rb

```diff
@@ -5,7 +5,7 @@ def update_from_api(invoice_id:, params:)
     invoice = Invoice.find_by(id: invoice_id)
 
     return result.not_found_failure!(resource: 'invoice') if invoice.blank?
-    return result.fail!(code: 'invalid_status') unless valid_status?(params[:status])
+    return result.single_validation_failure!(field: :status, error_code: 'value_is_invalid') unless valid_status?(params[:status])
```

Comment:
Style/IfUnlessModifier: Modifier form of `unless` makes the line too long. (https://rubystyle.guide#if-as-a-modifier)